PYI-285: Create minimal lambda and service class for returning the user info

dannyhayes-gds opened the pull request:
Co-authored-by: Sunil Dussoye <sunil.dussoye@digital.cabinet-office.gov.uk>

<!-- Provide a general summary of your changes in the Title above -->
<!-- Include the Jira ticket number in square brackets as prefix, eg `[P4-XXXX] PR Title` -->

## Proposed changes

### What changed
Created a new minimal lambda that accepts an Access token and returns the protected user info resource.
<!-- Describe the changes in detail - the "what"-->

### Why did it change
Required as a part of the minimal oauth flow where a user can use an access token in order to retrieve the protected user info response
<!-- Describe the reason these changes were made - the "why" -->

### Issue tracking
<!-- List any related Jira tickets or GitHub issues -->
<!-- List any related ADRs or RFCs -->
<!-- Delete/copy as appropriate -->

- [PYI-285](https://govukverify.atlassian.net/browse/PYI-285)

## Checklists

### Environment variables or secrets

<!-- Delete if changes DO include new environment variables or secrets -->
- [x] No environment variables or secrets were added or changed

dannyhayes-gds committed:
PYI-285: Create minimal lambda and service class for returning the user info

Co-authored-by: Sunil Dussoye <sunil.dussoye@digital.cabinet-office.gov.uk>

instantiator commented on src/main/java/uk/gov/di/ipv/domain/ErrorResponse.java:
```diff
@@ -8,7 +8,9 @@
     ERROR_1000(1000, "Missing query parameters for auth request"),
```
I'd suggest renaming these enum values so that they're easy to choose and understand from where they're used, eg.

```
MissingQueryParameters(1000, "Missing query parameters for auth request"),
MissingRedirectURI(1001, "Redirect URI is missing from auth request"),
```
etc.

dannyhayes-gds replied:
Yeah that's a good shout. I'll change all the enum values

idrop commented on src/main/java/uk/gov/di/ipv/service/UserInfoService.java:
```diff
@@ -0,0 +1,52 @@
+        userInfo.put("aud", ORCHESTRATOR_URN);
+        userInfo.put("sub", ORCHESTRATOR_URN);
+        userInfo.put("identityProfile", identityProfile);
+        userInfo.put("requestedLevelOfConfidence", ConfidenceLevel.MEDIUM);
```
Could you consider:
````
       Map<String, Object> userInfo = Map.of(
                "iss", ISSUER_URN,
                "aud", ORCHESTRATOR_URN,
                "sub", ORCHESTRATOR_URN,
                "identityProfile", identityProfile,
                "requestedLevelOfConfidence", ConfidenceLevel.MEDIUM);
```` 
which makes the map unmodifiable? Slightly pedantic for my first review I grant you :)

dannyhayes-gds replied:
Yeah not a problem, I'll add this in just now.

dannyhayes-gds committed:
Address PR comments by updating the ErrorResponse names with descriptive values and update how the userInfo map object is constructed.

Also update the error handling of the access token handler to make use of the ErrorResponses enum.

dannyhayes-gds committed:
Removed GPG45 domain classes
